packages/app/src/cli/services/dev/extension/utilities.ts

```diff
@@ -23,5 +23,33 @@ export async function getCartPathFromExtensions(
  * Returns the surface for UI extension from an extension point target
  */
 export function getExtensionPointTargetSurface(extensionPointTarget: string) {
-  return extensionPointTarget.toLowerCase().replace(/(::|\.).+$/, '')
+  const domain = extensionPointTarget.toLowerCase().replace(/(::|\.).+$/, '')
+
+  switch (domain) {
+    // Covers Checkout UI extensions and Post purchase UI extensions (future)
+    case 'purchase': {
+      const page = extensionPointTarget.split('.')[1]
+
+      if (page === 'post') {
+        return 'post_purchase'
+      }
+
+      // Checkout UI extensions
+      return page
```

Comment:
Checkout's renamed targets have this structure `purchase.checkout.actions.render-before` so we need to return the second portion.

cc @kumar303 @jamesvidler to double check.

Reply:
@vividviolet confirmed, looks like this being parsed correctly 👍 .

![Screenshot 2023-07-18 at 5 35 32 PM](https://github.com/Shopify/cli/assets/12719665/016730f8-4bc9-478b-8770-202c8fc72fdd)